processing/src/main/java/org/apache/druid/segment/loading/MMappedQueryableSegmentizerFactory.java

```diff
@@ -34,9 +35,29 @@
  */
 public class MMappedQueryableSegmentizerFactory implements SegmentizerFactory
 {
+  /**
+   * A static method to make a segmentizer factory that can be serialized by Jackson.  This exists because the object
+   * doesn't actually need the IndexIO object in order to be serialized by Jackson, but the public constructor
+   * *does* require it.  We leave the public constructor alone because if indexIO is null, this SegmentizerFactory
+   * is largely useless, so we just create this one static method to enable creating the object for this singular
+   * use case.
+   *
+   * @return a SegmentizerFactory that can be used to be serialized by Jackson
+   */
+  @SuppressWarnings("unused")
+  public static MMappedQueryableSegmentizerFactory makeForSerialization()
```

Comment:
nit: shouldn't this just be added for whatever needs it?

Reply:
Removed it from this PR.